enhancements/bundle-validation.md

```diff
@@ -0,0 +1,220 @@
+
+As a community contributor to the bundle validation library, I would want to understand the workflow of contributing to the core validation strategies or customized validations as part of an extended package.
+
+### Implementation Details/Notes/Constraints
+- The bundle validation library should list out errors based on static scanning of a bundle directory and locate their root causes if applicable. 
+
+- Much like project [flak8](https://pypi.org/project/flake8/) where a [list of validations](https://www.flake8rules.com/) can be individually enabled/disabled, individual bundle validating rule should also be separable. Ultimately, tools that are interested in importing the validation library should be able to cherry-pick validations used for different scenarios and configurations. 
+
+- The library should provide all validations currently available within the operator-courier.
+
+- The bundle validation library will be used to validate OLM/registry manifests and expose an interface to add validators for any scheme. 
+
+- Errors and alerting messages should be formatted in a standardized fashion.
+
+- The structure of the validation library needs to be well-organized and defined in a README.md to encourage community contribution.
+
+- The library should allow customized validation strategies as part of an extended package.
+
+- The library could potentially group different sets of validations to support specific bundle types or targeted cluster settings, ie: "validations for Kube 1.18", etc.
+
+#### Initial set of validations
+
+There exist validation steatagies avaliable in _operator-courier_ and _api_ repositories which would serve as an initial set of validations:
+
+- Validate the metadata for every CRD that exists in a bundle.
+    - E1: crd metadata not defined
+    - E2: crd metadata.name not defined
+    - E3: crd apiVersion not defined
+    - E4: crd spec not defined
+    - E5: crd spec.names not defined
+    - E6: crd spec.names.kind not defined
+    - E7: crd spec.names.plural not defined
+    - E8: crd spec.group not defined
+    - E9: crd spec.version pr spec.versions not defined
+    - E10: error converting versioned crd to unversioned crd
+    - E11: invalid CRD based on "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation"
+
+
+- Ensure the CSV and its metadata contain required fields, installable as deployemnts, and all its CRDs has specified owner(s).
+    - E12: csv metadata not defined
+    - E13: csv metadata.name not defined
+    - E14: csv name must have format: {operator name}.(v)X.Y.Z
+    - E15: csv name contains an invalid semver
+    - W1: csv metadata.annotations item is not defined if is not ["categories", "description", "containerImage", "createdAt", "support"]
+    - W2: csv metadata.annotations.certified not defined.
```

Comment:
This is an operatorhub.io requirement - should probably be done in an extended package.